packages/api/src/controllers/taskController.js

```diff
@@ -58,6 +62,52 @@ async function getTaskAssigneeAndFinalWage(farm_id, user_id, task_id) {
   return { assignee_user_id, finalWage };
 }
 
+async function formatAnimalMovementTaskForDB(data) {
+  if (data.locations && data.locations.length > 1) {
+    throw customError('Only one location can be assigned to this task type', 400);
+  }
+
+  if (!data.animal_movement_task) {
+    return data;
+  }
+
+  if (data.animal_movement_task.purpose_relationships) {
+    throw customError(
+      `Invalid field: "purpose_relationships" should not be included. Use "purposes" instead`,
+    );
+  }
+
+  if (!data.animal_movement_task.purposes) {
+    delete data.animal_movement_task.purposes;
+    return data;
+  }
+
+  checkIsArray(data.animal_movement_task.purposes, 'purposes');
```

Comment:
I think these kind of checks would be better placed in the middleware, in the `checkAnimalTask` function along with the checks you added there for the other fields. There aren't any custom entities being inserted/handled here like it happens for other tasks, so I think it wouldn't be correct to add this to `checkCustomDependencies`